### ISSUE TITLE
Missing symlinks 

### ISSUE THREAD

**Issue by Bayharbourbutcher1** [sureshfizzy/CineSync#40]
Cinesync keeps creating and then deleting symlinks for some reason, all the original links still exist in the torbox rclone mount. Torbox mount has not gone down during all this but CS for some reason decided to nuke all symlinks for me. Unsure, why this is happening!

**Comment by sureshfizzy** [comment]
Whats the exact situation here, i beleive it must have something to do with torbox only. Symlinks will be deleted only when the mount is not reachable and rclone mount is set to false.

**Comment by Bayharbourbutcher1** [comment]
Torbox is running in a docker container and has been up for several days, nothing in torbox logs to suggest it was down. Checked all links are showing in TB mount, not sure why CS thought mount was unavailable.

**Comment by Bayharbourbutcher1** [comment]
If i set rclone mount to true, i have errors in torbox rclone because you cannot write version.txt to its mount, its ready only therefore i am not using the mount validation variable.

**Comment by sureshfizzy** [comment]
> If i set rclone mount to true, i have errors in torbox rclone because you cannot write version.txt to its mount, its ready only therefore i am not using the mount validation variable.

That has been fixed, but another problem on torbox is, the mount is very poor. So the rclone mount will not work as expected only with torbox.


Regarding the deletion, I'm afraid i couldn't dig into that deeper without reproducing it. How frequently and how good is this for reproducing ? Because, it has been running without any problem with other mounts.

**Comment by Bayharbourbutcher1** [comment]
Yeah i agree their rclone mount doesn't work as well as real debrid. So last night, i updated and tried CS again and i had like 120 movies and shows in TB mount, CS started creating symlinks and when i woke up this morning there were no symlinks in the CS destination folder. All files in TB mount are still playable.

**Comment by sureshfizzy** [comment]
When the symlink is deleted, does it recreate it automatically again or you have to restart the container ?

**Comment by Bayharbourbutcher1** [comment]
Currently there are no symlinks in the CS folders, if i stop and restart container i am sure it will start recreating them. In fact ill try it now and let you know

**Comment by sureshfizzy** [comment]
> Currently there are no symlinks in the CS folders, if i stop and restart container i am sure it will start recreating them. In fact ill try it now and let you know
> 
> Edit: Stopped and restarted CS container, its currently creating symlinks again.

When this happens next time, do not restart cinesync, instead just go inside the container and do df -h and see if you can find torbox inside it.

**Comment by Bayharbourbutcher1** [comment]
Ok, no worries. Ill do that and let you know

**Comment by Bayharbourbutcher1** [comment]
Happened again last night, torbox folder did exist inside the CS container and so did all the files in TB mount but for some reason CS decided to delete both movies and shows symlink folders.

**Comment by sureshfizzy** [comment]
> Happened again last night, torbox folder did exist inside the CS container and so did all the files in TB mount but for some reason CS decided to delete both movies and shows symlink folders.

Seems somehow the torbox mount is unstable at this point, just enable rclone mount flag to prevent deletion.

**Comment by Bayharbourbutcher1** [comment]
I'll test it with rclone flag first and let you know

**Comment by digi-banana** [comment]
I'm having the same issue with real debrid. Rclone mount flag has always been set to true, but it's been deleting the symlinks.

Is there a way to just completely disable symlink cleaning? It's pretty rare for someone to delete things from real debrid anyway, so I'm not sure if it's a very meaningful function

**Comment by sureshfizzy** [comment]
> I'm having the same issue with real debrid. Rclone mount flag has always been set to true, but it's been deleting the symlinks.
> 
> Is there a way to just completely disable symlink cleaning? It's pretty rare for someone to delete things from real debrid anyway, so I'm not sure if it's a very meaningful function

Can you share me logs to see what is happening over here ? I'm currently working with ui and improving backend for accuracy for v3.0.

Also does your system restart during this process ?

**Comment by digi-banana** [comment]
Here's the last 1,000 lines in the log file:

https://pastebin.com/gGeQTDaN

Didn't restart the system. I was running it for the last couple of days and it was rebuilding my Plex library, but when I checked it this morning, a lot of files were missing, and it continued to delete more and more, so I ended up turning the container off.

**Comment by digi-banana** [comment]
Awesome! Thank you so much for your work on this!!
